packages/note/src/react/index.tsx

```diff
@@ -0,0 +1,235 @@
+    ),
+  metadata: (themeName: ValueOf<typeof themeNames>) =>
+    compose(
+      css(stylesheet['.psds-note__metadata']),
+      css(stylesheet[`.psds-note__metadata.psds-theme--${themeName}`])
+    ),
+  metadataDatum: (themeName: ValueOf<typeof themeNames>) =>
+    compose(
+      css(stylesheet['.psds-note__metadata-datum']),
+      css(stylesheet[`.psds-note__metadata-datum.psds-theme--${themeName}`])
+    ),
+  metadataDot: () => css(stylesheet['.psds-note__metadata-dot'])
+}
+
+interface NoteProps extends React.HTMLAttributes<HTMLDivElement> {
+  actionBar?: React.ReactElement<typeof Action>[]
+  actionBarVisible?: boolean
+  avatar?:
+    | React.ReactElement<typeof Avatar>
+    | React.ReactElement<typeof AvatarLink>
+  heading?: string | React.ReactElement
+  message: React.ReactNode
+  metadata?: React.ReactNode[]
+  onMouseOut?: React.MouseEventHandler
+  onMouseOver?: React.MouseEventHandler
+}
+interface NoteStatics {
+  Action: typeof Action
+  AvatarLink: typeof AvatarLink
+  List: typeof NoteList
+}
+
+const Note: React.FC<NoteProps> & NoteStatics = props => {
+  const {
+    actionBar,
+    actionBarVisible,
+    avatar,
+    heading,
+    message,
+    metadata,
+    onMouseOut,
+    onMouseOver,
+    ...rest
+  } = props
+  const themeName = useTheme()
+  const [isHovered, setIsHovered] = React.useState(false)
+
+  const handleMouseOver = (evt: React.MouseEvent) => {
+    setIsHovered(true)
+    if (onMouseOver) onMouseOver(evt)
+  }
+
+  const handleMouseOut = (evt: React.MouseEvent) => {
+    setIsHovered(false)
+    if (onMouseOut) onMouseOut(evt)
+  }
+
+  const hasActions = Array.isArray(actionBar) && actionBar.length > 0
+  const hasAside = !!avatar
+  const hasHeading = !!heading
+  const hasMetadata = Array.isArray(metadata) && metadata.length > 0
+
+  const renderActionBar = hasActions ? (
+    <div
+      {...styles.actionBar(themeName, {
+        hasHeading,
+        hasMetadata,
+        actionBarVisible: actionBarVisible || isHovered
+      })}
+    >
+      {(actionBar || []).map((action, key) => {
+        return React.cloneElement(action, { key })
+      })}
+    </div>
+  ) : null
+
+  return (
+    <div
+      {...styles.note(themeName)}
+      {...rest}
+      onMouseOver={handleMouseOver}
+      onMouseOut={handleMouseOut}
+    >
+      {hasAside && (
+        <div {...styles.aside()}>
+          {React.cloneElement(avatar as React.ReactElement<any>, {
+            size: avatarSizes.xSmall
+          })}
+        </div>
+      )}
+
+      <div {...styles.contents(themeName)}>
+        {hasHeading && (
+          <div {...styles.header()}>
+            {heading && <div {...styles.heading()}>{heading}</div>}
+            {renderActionBar}
+          </div>
+        )}
+
+        {message}
+
+        <div {...styles.footer()}>
+          {hasMetadata && (
+            <div {...styles.metadata(themeName)}>
+              {(metadata || []).map((datum: React.ReactNode, i) => {
```

Comment:
maybe move the default array to the props destructing...your call

Reply:
I don't want a default array, I think.  This is just because TS isn't smart enough to see the safety of `hasMetadata` above.